crates/shared/src/http_client.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::{anyhow, Result};
+use reqwest::Response;
+
+/// Extracts the bytes of the response up to some size limit.
+///
+/// Returns an error if the byte limit was exceeded.
+pub async fn response_body_with_size_limit(
+    response: &mut Response,
+    limit: usize,
+) -> Result<Vec<u8>> {
+    let mut bytes = Vec::new();
```

Comment:
I think it would make sense to offer a high performance API which allows you to pass in the buffer to avoid copying the data out of the function.
Then `response_body_with_size_limit()` could be a simple wrapper function around that more performant API. I believe if `response_body_with_size_limit()` is tagged with `#[inline]` we should have a simple API while still avoiding unnecessary copies.

Reply:
There is no copying going on when returning the data out of the function. The data lives on the heap and Vec stores a pointer to it and the size so it's like returning two integers. For high performance passing in the buffer would still be advantageous because it allows you to reuse the allocation in the future but I don't want to make this more complicated before specifically wanting to use that.

Reply:
Yeah, of course. You're right. 🤦‍♂️ 
It's all good then. 👍